lib/elixir/test/elixir/option_parser_test.exs

```diff
@@ -108,6 +108,22 @@ defmodule OptionParserTest do
            == {[no_value: true], [], []}
   end
 
+  test "parses configured counters" do
+    assert OptionParser.parse(["--verbose"], switches: [verbose: :count])
+           == {[verbose: 1], [], []}
+    assert OptionParser.parse(["--verbose", "--verbose"], switches: [verbose: :count])
+           == {[verbose: 2], [], []}
+    assert OptionParser.parse(["--verbose", "-v", "-v", "--", "bar"], aliases: [v: :verbose], strict: [verbose: :count])
+           == {[verbose: 3], ["bar"], []}
+  end
+
+  test "parses configured counters over integers" do
+    assert OptionParser.parse(["-v", "-v"], aliases: [v: :verbose], strict: [verbose: [:count, :integer]])
+           == {[verbose: 2], [], []}
+    assert OptionParser.parse(["--verbose", "-v", "-v"], aliases: [v: :verbose], strict: [verbose: [:integer, :count]])
+           == {[verbose: 3], [], []}
+  end
```

Comment:
@lexmag I have found the way to support both `[:count, :integer]` and `:count` over `:integer` so is this test case should have covered